blades/combo_blade.h

```diff
@@ -0,0 +1,143 @@
+#define BLADES_COMBO_BLADE_H
+
+/*
+Usage: ComboBlade(blade1, blade2)
+This allows multiple blades to be combined into a single linear blade even if
+they use different data lines for control.
+
+The resulting blade will have a length of the number of LEDs in the first and
+second blade combined. When the style is running, the second blade's LEDs will
+be controlled as though they were in line and after the first blade's LEDs
+
+For a simple example, if you have a saber with 2 crystal LEDs running on different
+data lines rather than in line with one another, you might often do the following.
+
+{ 0,
+  WS281XBladePtr<128, bladePin, Color8::GRB, PowerPINS<bladePowerPin2, bladePowerPin3> >(),
+  WS281XBladePtr<1, blade4Pin, Color8::GRB, PowerPINS<bladePowerPin2, bladePowerPin3> >(),
+  WS281XBladePtr<1, blade3Pin, Color8::GRB, PowerPINS<bladePowerPin2, bladePowerPin3> >(),
+}
+
+The above way allows us to assign either the same style or different styles to
+both LEDs
+
+However, if we assign the same style the LEDs will do the same thing since they
+are both seen as the first LED in a 1 LED longht blade. To avoid this, we can
+use ComboBlade, which will result in the style being run against an intermediate
+blade with a length of 2, rainbow and stripe styles to move between the two
+
+{ 0,
+  WS281XBladePtr<128, bladePin, Color8::GRB, PowerPINS<bladePowerPin2, bladePowerPin3> >(),
+  ComboBlade(WS281XBladePtr<1, blade4Pin, Color8::GRB, PowerPINS<bladePowerPin2, bladePowerPin3> >(),
+             WS281XBladePtr<1, blade3Pin, Color8::GRB, PowerPINS<bladePowerPin2, bladePowerPin3>>()),
+
+In the above example we must reduce the NUM_BLADES by 1 since the ComboBlade itself
+is the only thing that counts as an entry in the blade array.
+
+*/
+
+class ComboBladeWrapper : public BladeWrapper, BladeStyle {
+public:
+  ComboBladeWrapper(BladeBase* blade1, BladeBase* blade2):
+    blade1_num_leds_(blade1->num_leds()),
+    blade2_(blade2),
+    real_style_(nullptr)
+  {
+    blade_ = blade1;
+  }
+
+  // BladeStyle implementation
+  void activate() override {
+    real_style_->activate();
+  }
+  void deactivate() override {
+    real_style_->deactivate();
+  }
+  void run(BladeBase* blade) override {
+    if (blade == blade_) {
+      real_style_->run(this);
+    }
+  }
+
+  bool NoOnOff() override { return real_style_->NoOnOff(); }
+
+  bool Charging() override { return real_style_->Charging(); }
+
+  bool IsHandled(HandledFeature effect) override {
+    if (real_style_)
+      return real_style_->IsHandled(effect);
+    return false;
+  }
+
+  OverDriveColor getColor(int i) override { return real_style_->getColor(i); }
+
+  int get_max_arg(int arg) override { return real_style_->get_max_arg(arg); }
+
+  // BladeBase implementation
+  int num_leds() const override {
+    return blade1_num_leds_ + blade2_->num_leds();
+  }
+  void set(int led, Color16 c) override {
+    if (led < blade1_num_leds_) {
+      return blade_->set(led, c);
```

Comment:
Now that I think about it, this doesn't really work.
In most cases, blades don't allocate space for pixel colors until right before the run() call. This means that calling set() on a blade other than the one that called run() is not allowed. Either this class would need to buffer the required pixels, or there would need to be some major rewrites to how blades work.

Sorry it took me so long to realize this. I didn't mean to make you do all the work to clean this up only to have it not checked in....

Reply:
No worries, that makes sense. I put together a version that caches the results for blade2 and it seems to work as well, though it does mean more dynamic allocation unfortunately (only alternative seemed to be statically allocating more memory than needed).

Sorry for silence on my end, lots going on the last couple weeks so haven't been able to get back to this

Reply:
The allocation would only happen for those people who actually use this kind of setup, so I think that is not unreasonable.